### PR TITLE
[merge] Add Fabric V1 packaging capabilities

### PR DESCRIPTION
**Pull request by james-wallis**
### Summary
* Add Fabric V1 capabilities into the merge branch alongside V2
* To determine the version it will ask the user using an input box but for packaging via the deploy view the version can be passed in

Afaik V1 doesn't have the metadata capabilities so I've made it throw an error

### Testing
* Updated and added unit tests
* Updated and added cucumber tests for packaging

Signed-off-by: James Wallis <j@wallis.dev>

### CODE REVIEW

File: packages/blockchain-extension/cucumber/helpers/smartContractHelper.ts [hunk 1]
```diff
@@ -123,7 +123,7 @@ export class SmartContractHelper {
         return contractDirectory;
     }
 
-    public async packageSmartContract(name: string, version: string, language: string, directory: string): Promise<PackageRegistryEntry> {
+    public async packageSmartContract(name: string, version: string, language: string, directory: string, fabricVersion: any): Promise<PackageRegistryEntry> {
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/cucumber/helpers/smartContractHelper.ts:126]: Can we give `fabricVersion: any` a proper type and also make it possible so we don't have to do `const sanitisedFabricVersion: number = (fabricVersion === 1 || fabricVersion === 'v1' || fabricVersion === '1') ? 1 : 2;`.
I think it would be good to standardise this function so we can only call it one way.

Comment (lquintai) [reply packages/blockchain-extension/cucumber/helpers/smartContractHelper.ts:126]: We already have `UserInputUtil.V1_4_2` and `UserInputUtil.V2_0`, could we use them here?

Comment (james-wallis) [reply packages/blockchain-extension/cucumber/helpers/smartContractHelper.ts:126]: I can remove the `fabricVersion === 1` as that shouldn't be hit but I'll leave `'v1's and `'1'` as it acts as a failsafe for the cucumber tests so that they work whether we put "Fabric 1" or "Fabric v1".

File: packages/blockchain-extension/cucumber/data/cucumber.code-workspace [hunk 2]
```diff
@@ -3,6 +3,38 @@
 		{
 			"path": "emptyProject",
 			"name": "emptyProject"
+		},
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/cucumber/data/cucumber.code-workspace:6]: Can you undo these changes please

Comment (james-wallis) [reply packages/blockchain-extension/cucumber/data/cucumber.code-workspace:6]: Sure

**Review by Jakeeyturner — APPROVED** [review]
